### DISCUSSION TITLE
border-x and border-y

### DISCUSSION THREAD

**Discussion by jenstornell** [tailwindlabs/tailwindcss#2688] [Ideas]
We already have `mx-1` and `my-1` for horizontal and vertical margins. The same thing for paddings, but not for border.

## Current way

```html
<div class="border-l border-r border-gray-300"></div>
```

## Suggested syntax

```html
<div class="border-x border-gray-300"></div>
```

**Comment by simonswiss** [comment]
That's an interesting take! 

From personal experience, I very often want the same spacing applied vertically or horizontally on elements, but cannot recall one time where I wanted an element to have vertical or horizontal borders defined together specifically.

I either go with all borders, or only one (thick) border for an alert/notice type of UI element.

I am curious - what are some of the scenarios where you define borders specifically for top/bottom or left/right?

I would _anticipate_ that there is not enough common use cases to justify adding these classes to Tailwind core, and this would probably be better suited to a plugin implementation.

If you factor in the border width, responsive, variants etc, that would generate a non-neglectable amount of utilities to the compiled styles.

**Reply by jenstornell** [reply]
I think I had an active tab that needed borders on both sides and on top.

I did not read the docs and simply expected `border-x` to work because it works with padding and margins. The units are not always consistant.

I understand your stand, that it's not a common case.

**Reply by jenstornell** [reply]
Today I'm at it again. Here is a screenshot where it could be helpful with `border-y`. Not a very common use case maybe.

![image](https://user-images.githubusercontent.com/927011/97709910-5b6cde00-1abb-11eb-9d78-ca7a97578d56.png)

**Reply by y1n0** [reply]
they are very useful indeed. in my case I need to use them along with `.divide-x` and .`divid-y` to make a list of elements have a start and end borders too.

![list start/end borders with .border-y and .divide-y](https://user-images.githubusercontent.com/8240619/113570004-5b917c80-960b-11eb-937e-f9c0c6d4e606.png)

**Reply by simonswiss** [reply]
The JIT implementation means that the "bloat" argument can be discarded to an extent. 

If there is sufficient demand for it, I can imagine `border-x` and `border-y` utilities have a place in Tailwind core at some point down the road. There are no concrete plans to add this as of now, so not making any promises here 😅

**Comment by jenstornell** [comment]
Because of the new version this issue will be solved. Thanks!

https://github.com/tailwindlabs/tailwindcss/discussions/5668
